lib/tapioca/compilers/dsl/identity_cache.rb

```diff
@@ -177,7 +178,63 @@ def create_fetch_by_methods(field, klass, constant)
               class_method: true,
               parameters: parameters,
               return_type: "::#{constant}"
+            ) if constant.respond_to?("#{name}!".to_sym)
+
+            klass.create_method(
+              name,
+              class_method: true,
+              parameters: parameters,
+              return_type: "T.nilable(::#{constant})"
+            ) if constant.respond_to?(name.to_sym)
+          elsif constant.respond_to?(name.to_sym)
+            klass.create_method(
+              name,
+              class_method: true,
+              parameters: parameters,
+              return_type: COLLECTION_TYPE.call(constant)
             )
+          end
+
+          if field_length == 1
+            name = "fetch_multi_by_#{fields_name}"
+
+            if constant.respond_to?(name.to_sym)
+              klass.create_method(
+                name,
+                class_method: true,
+                parameters: [
+                  Parlour::RbiGenerator::Parameter.new("index_values", type: "T.untyped"),
+                  Parlour::RbiGenerator::Parameter.new("includes:", default: "nil", type: "T.untyped"),
+                ],
+                return_type: COLLECTION_TYPE.call(constant)
+              )
+            end
+          end
+        end
+
+        sig do
+          params(
+            field: T.untyped,
+            klass: Parlour::RbiGenerator::Namespace,
+            constant: T.class_of(::ActiveRecord::Base),
+          ).void
+        end
+        def create_aliased_fetch_by_methods(field, klass, constant)
```

Comment:
We shouldn't really need a new method, right? Since the library just generates the same kind of attributes for both the `cache_index` and `cache_attribute` calls. We just weren't handling the aliases set in those attributes properly before, from what I understand.

Reply:
It was not just a matter of handling the alias name, we were ignoring all methods created by `cache_attribute_by_alias`.

`cache_index` generates extra methods [here](https://github.com/Shopify/identity_cache/blob/01e90ed9fffe960ff96e9e54eaa7a03fa47a4d85/lib/identity_cache/with_primary_index.rb#L62-L92) which are not generated by `cache_attribute`. These are  the ones we were already generating in the RBIs.

However, we were completely ignoring the methods generated by `cache_attribute_by_alias` [here](https://github.com/Shopify/identity_cache/blob/771b0f98324274c8fd79bf2d2094ca9545f3af45/lib/identity_cache/cached/attribute_by_one.rb#L16-L25).

Also, the extra methods created by `cache_index` have a different signature.

Reply:
Ok, in that case, this method should generate only the methods that `cache_attribute_by_alias` would have generated and nothing more.